gcstaublin/pregame
# Svg

## What this did

Proof of concept for a SVG workflow. When you type `gulp svg`, all svg located inside the `icons` folder will be combined into one file and inject into the html document, just above the opening `<body>` tag.

Then inside your html you call svgs using the ID of that icon like so:

```
<span class="icon__container space-right--sm">
  <svg aria-hidden="true" class="icon icon--sm">
    <use xlink:href="#icons_upload"></use>
  </svg>
  <span class="assistive-text">Text goes here</span>
</span>
```

## How I did this
I took 10 icons from the master SVG sheet and normalized their size to a consistent canvas. I actually copied each of these icons into Illustrator, then exported each and put them into the `icons` folder.

The gulp task uses `gulp-svgstore`, `gulp-inject`, 'gulp-cheerio` to combine the svgs and inject them into the html.

## To test this out
Install the gulp dependencies
`npm install`

Run the gulp task
`gulp svg`

NOTE: The SVG sprite will already exist inside the html document. So you might experiment with adding another icon to the icons folder, as well as to your html to see the magic happen.

## Questions
  1. How would this work inside an actual build process?
  1. Rather than having ALL icons injected into our documents, which would become unwieldy, how can we make separate sprite that would work for needs of different clusters. Unfortunately, linking to external files, an SVG file outside the document, doesn't work in any versions of IE.
  1. This assumes SVGs that are normalized. How do we get there?
  1. Other questions.
